spec/features/admin_adds_video_spec.rb

```diff
@@ -3,46 +3,68 @@
 feature "Admin adds video" do
   given(:admin) { Fabricate(:admin) }
   given(:user) { Fabricate(:user) }
-
-  background do
-    Fabricate(:category, title: 'comedy')
-    Fabricate(:category, title: 'action')
-  end
+  given(:comedy) { Fabricate(:category, title: 'comedy') }
+  given(:up) { { title: 'Up',
+                 category: comedy.title.capitalize,
+                 description: 'A funny movie',
+                 large_cover: 'spec/support/uploads/up_l.jpg',
+                 small_cover: 'spec/support/uploads/up_s.jpg',
+                 video_url: 'http://clips.vorwaerts-gmbh.de/big_buck_bunny.mp4'} }
+  given(:invalid_video_input) {{ category: comedy.title.capitalize,
+                                 description: 'This is the only input user made.' }}
 
   scenario "admin signs in and add video with correct input" do
     sign_in(admin)
 
-    visit new_admin_video_path
-    fill_in "Title", with: 'Up'
-    select('Comedy', :from => 'Category')
-    fill_in "Description", with: 'A funny movie.'
-    attach_file('Large cover', 'spec/support/uploads/up_l.jpg')
-    attach_file('Small cover', 'spec/support/uploads/up_s.jpg')
-    fill_in "Video URL", with: 'http://clips.vorwaerts-gmbh.de/big_buck_bunny.mp4'
-    click_button 'Add Video'
+    admin_adds_new_video(up)
 
-    expect(page).to have_content "You've added video Up."
-    expect(page).to have_content "Add a New Video"
+    check_if_video_added_and_redirect_back_to_add_new
+
+    sign_out(admin)
 
-    sign_out
     sign_in(user)
 
-    click_link "Videos"
-    expect(page).to have_selector("img[src='/uploads/up_s.jpg']")
-    click_video_on_home_path(Video.last)
-    expect(page).to have_selector("img[src='/uploads/up_l.jpg']")
-    expect(page).to have_selector("a[href='http://clips.vorwaerts-gmbh.de/big_buck_bunny.mp4']")
+    check_if_new_added_video_show_up_on_home(up)
+
+    check_if_new_added_video_has_correct_large_cover_and_video_url(up)
```

Comment:
For example this 'and' is to check the pages show the images correctly, so that is the same check, no need to split.

But checking the "flash[:success]" and checking redirect is two things, better split them.

The abstraction depends on the target that I intend to checks, is it?